src/lib/locale/locales.js

```diff
@@ -65,12 +65,50 @@ function loadLocale(name) {
 // no arguments are passed in, it will simply return the current global
 // locale key.
 export function getSetGlobalLocale (key, values) {
-    var data;
+    var data, userLanguage;
+
+    // eval is the only way we can check vaule of global variable
+    var isBrowser = new Function('try {return this === window;}catch(e){ return false ;}');  // jshint ignore:line
+
+    if (isBrowser()) {
+        userLanguage = [
+            window.navigator.userLanguage,
+            window.navigator.language
+        ];
+    } else {
+        userLanguage = ['en'];
+    }
+
+    // in case userLanguage or language is undefined
+    userLanguage = userLanguage.filter(function noUndefined(value) {
+        return value !== undefined;
+    });
+
+    // in case userLanguage and language are undefined at the
+    // same time
+    if (userLanguage.length === 0) {
+        userLanguage = ['en'];
+    }
+
     if (key) {
+        if (isArray(key)) {
```

Comment:
This logic is redundant, the case that the key is an array is already handled by `chooseLocale` (which is called by `getLocale`.

Additionally, if `chooseLocale` does not know about the `'default'` locale, then `moment().locale('default')` is going to fail.